frontend/model/contracts/identity.js

```diff
@@ -283,8 +283,7 @@ sbp('chelonia/defineContract', {
           throw new Error(`Cannot leave group ${groupContractID} because the reference hash does not match the latest`)
         }
 
-        state.groups[groupContractID].hasLeft = true
-        delete state.groups[groupContractID].inviteSecret
+        state.groups[groupContractID] = { hash: reference, hasLeft: true }
```

Comment:
There might be other attributes on `groups` that you're overwriting here that shouldn't be overwritten.

Reply:
There aren't. This is when leaving a group, so we should keep the attributes there at a minimum.

Reply:
It's true that there aren't at the moment, but in the future if more attributes are added then developers must remember to update this code to make sure nothing breaks. Whereas the way it worked before didn't need as much attention.

Reply:
> but in the future if more attributes are added then developers must remember to update this code to make sure nothing breaks

Nothing should break because these are chatroom that we're not part of. Note that attributes are _already_ overwritten when _joining_ a chatroom. The only reason we keep these values around at all (before we deleted the whole thing) is for accounting purposes, so that re-joining works correctly.

> then developers must remember to update

I think this is a good thing. You shouldn't need attributes from chatroom you're not a part of. The only exception I can think of is for a future feature that allows you to 'block' a chatroom (to prevent re-joining), in which case it makes sense to be explicit.

Reply:
Either way, making things as they were before still means that developers need to pay attention to this. If a `foo` attribute were to be added, which is not used after having left, it needs to have a corresponding `delete` here.

Reply:
ok that last comment makes sense... can you add a comment here explaining that?